### PR TITLE
fix: clang-tidy-20 warnings

### PR DESCRIPTION
**Pull request by ckerr**
fix some clang-tidy-20 warnings in libtransmission.

### CODE REVIEW

File: libtransmission/peer-mgr-active-requests.cc
Comment (tearfur) [inline libtransmission/peer-mgr-active-requests.cc:117]: Not sure why the CI passed, but `std::map::contains()` is a C++20 feature.

Speaking of which, the C++20 bump has been brought up multiple times in the past year. We might as well make the jump now, then gradually start using its features depending on the CI results. wdyt?

File: libtransmission/peer-mgr-active-requests.cc
Comment (tearfur) [inline libtransmission/peer-mgr-active-requests.cc:160]: ditto

File: libtransmission/peer-mse.cc
Comment (tearfur) [inline libtransmission/peer-mse.cc:88]: This lambda is only used once, maybe we should just inline it.

**Comment by Coeur** [comment]
Please don't jump to C++20 before merging the reviewed and approved fixes first.
Refactoring like this should be done last, after the fixes ready-to-merge.
https://github.com/transmission/transmission/issues/6952

For instance, among those fixes is a data corruption fix which is top priority.

File: libtransmission/peer-mgr-active-requests.cc
Comment (ckerr) [reply libtransmission/peer-mgr-active-requests.cc:117]: > Not sure why the CI passed, but std::map::contains() is a C++20 feature.

Because this is a `small::map` instead of a `std::map` :smile_cat: 

> Speaking of which, the C++20 bump has been brought up multiple times in the past year. We might as well make the jump now, then gradually start using its features depending on the CI results. wdyt?

Personally I'd rather be using C++20 than 17. My only question is whether or not a good C++20 compiler exists on all the CI machines, e.g. the older Macs and the old distros in TeamCity's Sanity runs. But writing a PR to bump to 20 is an easy way to find out :smile_cat:

**Comment by ckerr** [comment]
This PR doesn't jump to C++20. The '20' in the title is only for clang-tidy version 20

File: libtransmission/peer-mse.cc
Comment (ckerr) [reply libtransmission/peer-mse.cc]: good call. inlined in 33f9a54f897d57fe094843fb4e96a942366efa44

**Review by nevack — APPROVED** [review]
LGTM

File: libtransmission/peer-mgr-active-requests.cc
Comment (tearfur) [reply libtransmission/peer-mgr-active-requests.cc:117]: > This is a `small::map` instead of a `std::map` which has `contains()` even in C++17

Oh.

> But writing a PR to bump to 20 is an easy way to find out 😸

Yep I'll be getting on that.

**Review by tearfur — APPROVED** [review]

File: libtransmission/peer-mgr-active-requests.cc
Comment (Coeur) [reply libtransmission/peer-mgr-active-requests.cc:117]: We don't have CI for macOS 11 anymore, and I'd like to still support it.
But it's your call: if we limit our support to what GitHub Actions can support, then what is the need for the TeamCity builds maintained by Mike?

Comment (Coeur) [reply libtransmission/peer-mgr-active-requests.cc:117]: Well, let's start with what we have: GitHub Actions support Xcode 13. Then it means that the C++20 features listed in this table with a minimum requirement of Xcode 14, 15 or 16 won't be supported:
https://developer.apple.com/xcode/cpp/#c++20

Comment (ckerr) [reply libtransmission/peer-mgr-active-requests.cc:117]: Let's move this discussion to https://github.com/transmission/transmission/pull/7191 :+1:

**Pull request merged**
